packages/entity-server/src/routes/hierarchy/relationships/middleware.ts

```diff
@@ -48,8 +48,10 @@ const getSubContext = (
   const field = (queryField ? extractFieldFromQuery(queryField) : baseField) || 'code';
   const filter = extractFilterFromQuery(allowedCountries, queryFilter);
   const { type, ...restOfFilter } = filter;
-  if (type === null || typeof type !== 'string') {
-    throw new Error(`${from}_filter:type must be a single, not null type constraint`);
+  if (type === null || (type !== undefined && typeof type !== 'string')) {
```

Comment:
Makes it a bit easier to see the allowed types:

```suggestion
  if (typeof type !== 'undefined' && typeof type !== 'string') {
```

Reply:
Oh yeah that's much better!